### DISCUSSION TITLE
How to set colorSpace in WebGPU？

### DISCUSSION THREAD

**Discussion by seven332** [emscripten-core/emscripten#22752] [Q&A] [answered]
WebGPU supports colorSpace in GPUCanvasContext.configure() method.
https://developer.mozilla.org/en-US/docs/Web/API/GPUCanvasContext/configure#colorspace

But emscripten doesn't seem to support it.
https://github.com/emscripten-core/emscripten/blob/63b8a7c9d8d0ebe15d7897a48b9fad34fad99026/src/library_webgpu.js#L2788

**Comment by sbc100** [comment]
@kainino0x

**Comment by kainino0x** [comment, marked as answer]
It is missing in Emscripten's version of webgpu.h right now. If you need it, we can patch it in, but otherwise it will be supported in some months when we complete the stabilization of webgpu.h and bring in the updated Emscripten bindings implementation (currently being worked on in Dawn as "emdawnwebgpu").

**Reply by seven332** [reply]
Thanks for the reply. I'm porting a design tool from WebGL to WebGPU which supports P3 color space. I would be grateful if the patch is available.

**Reply by kainino0x** [reply]
Thanks for letting me know. Unfortunately it will probably be a few weeks before I can work on it.
In the meantime it should be possible to call `configure()` from javascript directly, via `EM_ASM` and using the `jsvalstore` API ([see here](https://github.com/emscripten-core/emscripten/blob/main/test/webgpu_jsvalstore.cpp)) though note that it will break and be replaced by a similar-but-different API for js-wasm interop, along with all the rest of our breaking changes for webgpu.h.

**Reply by seven332** [reply]
Wow, EM_ASM + jsvalstore work like a charm.
